app/controllers/graphql_controller.rb

```diff
@@ -48,4 +59,34 @@ def handle_error_in_development(e)
 
     render json: { errors: [{ message: e.message, backtrace: e.backtrace }], data: {} }, status: 500
   end
+
+  def set_cache_fingerprints_for_records
+    Rails.cache.redis.pipelined do |redis|
+      records.each do |record|
+        redis.sadd(record.cache_key, fingerprint)
+      end
+    end
+  end
+
+   def fingerprint
+    @fingerprint ||= GraphQL::Query.new(
+      GraphqlCacheTestSchema,
+      query,
+      # context: context,
+      variables: variables,
+      operation_name: operation_name
+    ).fingerprint
+   end
+
+   def track_records
+    ActiveSupport::Notifications.subscribe('initialized.active_record') do |*args|
```

Comment:
do you need to unsubscribe after action?

Reply:
I don't know how subscriptions work under the hood, you're probably right. I wonder if the subscription will prevent garbage collecting the controller instance after the response is sent...